### PR TITLE
Switch from pkg_resources to importlib.metadata

### LINKED ISSUE #2116: RFE: Replace use of `pkg_resources` with `importlib.metadata`
See discussions:
[astropy/astropy#11091](https://github.com/astropy/astropy/pull/11091)
[pypa/pip#7413](https://github.com/pypa/pip/issues/7413)

```console
[tkloczko@devel-g2v SRPMS]$ rpm -ql python-pygments | grep py$ | xargs grep pkg_resources
/usr/lib/python3.8/site-packages/pygments/plugin.py:        import pkg_resources
/usr/lib/python3.8/site-packages/pygments/plugin.py:    return pkg_resources.iter_entry_points(group_name)
```

### PR DESCRIPTION
**Pull request by Jean-Abou-Samra**
importlib.metadata is a standard library replacement for
pkg_resources, available starting with Python 3.8. It is faster than
pkg_resources. This lets the plugin interface use it when available,
i.e. on Python 3.8 and later. On earlier Python, it uses the
importlib_metadata PyPI backport if available, and finally falls back
on pkg_resources. setup.cfg gains an extra called plugins, which can
allow a project to install "pygments[plugins]" in order to ensure that
plugins support is present even if installed with older Python
versions.

Timings for lexing an empty file with a lexer from a random plugin:

Before:

```
real	0m0,238s
user	0m0,210s
sys	0m0,029s
```

After:

```
real	0m0,141s
user	0m0,125s
sys	0m0,017s
```

Fixes #2116, #2126

### CODE REVIEW

**Comment by Jean-Abou-Samra** [comment]
Bah. This failed the CI initially because `entry_points()` didn't accept the `group` keyword argument. Turns out, it was added in Python 3.10 and in recent versions of the Python <3.8 backport. Of course, I had tested those two versions, and not 3.8 or 3.9…

File: pygments/plugin.py [hunk 1]
```diff
@@ -34,19 +36,34 @@
...
-
-    return pkg_resources.iter_entry_points(group_name)
+        from importlib.metadata import entry_points
+    except ImportError:
+        try:
+            from importlib_metadata import entry_points
+        except ImportError:
+            try:
+                from pkg_resources import iter_entry_points
+            except (ImportError, OSError):
```
Comment (Jean-Abou-Samra) [inline pygments/plugin.py:59]: Here I have a question. I kept this from the old code. Is there any particular reason an import might trigger `OSError`?

**Comment by birkenfeld** [comment]
Thanks for taking on this topic! I'll try to review when back home.

**Comment by Jean-Abou-Samra** [comment]
Thanks. No hurry, take your time.

File: pygments/plugin.py (hunk 1 above)
Comment (birkenfeld) [reply pygments/plugin.py:56]: I don't remember anything... possibly `pkg_resources` does some things already on import which might trigger this?

**Review by birkenfeld — APPROVED** [review]
LGTM, thanks!

File: pygments/plugin.py (hunk 1 above)
Comment (Jean-Abou-Samra) [reply pygments/plugin.py:56]: Ok, then I’ll stick with ImportError for the others, but keep OSError here to minimize risk.

**Pull request merged**